02.calendar/calendar.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+
+# コマンドラインオプション設定(デフォルト値は今日の年と月)
+params =ARGV.getopts("m:", "y:", "m:#{Date.today.mon}", "y:#{Date.today.year}")
+input_month = params["m"].to_i
+input_year = params["y"].to_i
+
+# 月の最初の日にちを取得
+start_date = Date.new(input_year, input_month, 1)
+
+# 月の最終日を取得
+end_date = Date.new(start_date.year, start_date.mon, -1)
+
+# 月と年を取得
+header = "#{start_date.mon}月 #{start_date.year}"
+
+# すべての曜日を取得
+week_days = ["日", "月", "火", "水", "木", "金", "土"]
+
+# カレンダーの最初の日までの余白を配列で作成
+margin_space = []
+start_date.wday.times do
+  margin_space.unshift("  ")
+end
+
+# すべての日付を配列で作成
+all_days = (start_date.day..end_date.day).to_a
+
+# 配列の要素を文字列型に変更、各要素を右寄せ
+# 1日ではないかつ、日曜日の日付の場合日付の前に改行文字を入れて改行
+calendar_formatted = all_days.map do |day|
+                       date = Date.new(start_date.year, start_date.mon, day)
```

Comment:
```suggestion
calendar_formatted = (start_date..end_date).map do |date|
```

のように書いた方がよりシンプルですね。

あと好みですが、僕だったらこんな感じでブロック内をインデントします。

```ruby
foo = bar.each do |b|
  # ...
  # ...
end
```

Reply:
修正しました！
https://github.com/jun-kondo/ruby-practices/pull/2/commits/5689a049fa61f969827adf0aa78d47ef37cd0850